src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessor.java

```diff
@@ -98,7 +101,16 @@ private <Result extends SearchPhaseResult> boolean shouldSkipProcessor(SearchPha
         }
 
         QueryPhaseResultConsumer queryPhaseResultConsumer = (QueryPhaseResultConsumer) searchPhaseResult;
-        return queryPhaseResultConsumer.getAtomicArray().asList().stream().filter(Objects::nonNull).noneMatch(this::isHybridQuery);
+        if (queryPhaseResultConsumer.getAtomicArray().asList().stream().filter(Objects::nonNull).noneMatch(this::isHybridQuery)) {
+            return true;
+        }
+        List<QuerySearchResult> querySearchResults = getQueryPhaseSearchResults(searchPhaseResult);
+        Optional<FetchSearchResult> fetchSearchResult = getFetchSearchResults(searchPhaseResult);
+        if (shouldSkipProcessorDueToIncompatibleQueryAndFetchResults(querySearchResults, fetchSearchResult)) {
+            log.debug("Query and fetch results do not match, normalization processor is skipped");
+            return true;
+        }
```

Comment:
is this the case when Query and Fetch results are not compatible but hybrid query clause is there?

if yes, then I think we should throw the exception rather than just doing pass through. This will lead users getting confused if normalization is working or not.

Reply:
Yes, at this moment we already checked if there is at least one result from hybrid query. 
Maybe I misunderstood the problem reported [in this issue](https://github.com/opensearch-project/neural-search/issues/497#issuecomment-1863499052). My understanding was that in such scenario with hybrid query there is a chance that fetch results have different number of docs comparing to query results. We currently are throwing exception in such case, like one that posted in that issue:

```
    "error": {
        "root_cause": [],
        "type": "search_phase_execution_exception",
        "reason": "The phase has failed",
        "phase": "query",
        "grouped": true,
        "failed_shards": [],
        "caused_by": {
            "type": "illegal_state_exception",
            "reason": "Score normalization processor cannot produce final query result"
        }
    },
    "status": 500
}
```
so the whole idea is still to throw an exception, just earlier?

Reply:
Yes, we are throwing exception but we don't know the root cause of the error. So when we do noop here, is the search getting failed going forward? When we were discussing this, I was under the impression that if there is some errors from fetch phase in that case we will fail after the processor execution. Is that happening  with this change?

Reply:
I tried multiple variations of possible errors in fetch phase, like explicitly throwing exception and updating collection of hits with irrelevant data. 
In case of exception system was failing but it was before even hitting our code. I think that correlates with your expectations, but that happened even without any change from our side. 
In case of inconsistent fetch and query data there were no exception and result was always updated to match results of fetch phase. Before this change there was an exception reported by user, with this change we're silently skipping normalization and results are returned to user